### ISSUE TITLE
Control AND/OR logic for Advanced Search

### ISSUE THREAD

**Issue by TauerOne** [vitmalina/w2ui#1990]
Is there a way to control the logic used for advanced search at run time so that, based on user input, I can programmatically set it to AND or OR?

**Describe the problem your feature will solve.**
User types "john" in the simple search box and searches, gets some results.  User opens the advanced search pop up and sees that all of the search fields are filled in with "john" (even integer and date fields).  User clicks the search button in the pop up and gets zero matches.  User is confused.  The issue is that the simple search box uses "OR" logic and the advanced search uses "AND" logic.

**Describe the solution you'd like**
Ideally, I would like something like a radio set at the bottom of the advanced search pop up that has two options such as "Match All" and "Match Any".

**Describe alternatives you've considered**
If there is a way to programmatically control the logic used by the advanced search, then I will find a way to prompt the user and set it accordingly. 

Thanks, and great product by the way.

**Comment by mpf82** [comment]
You could do the following:

```
    onSearch: function(event) {
      var obj = this;

      if (event.searchField === "multi") { // advanced search
        console.log("Advanced")
        obj.last.logic = 'AND'; // or 'OR', based on your use case
      } else { // simple search, either 'all' or a specific field
        console.log("Simple")
        obj.last.logic = 'OR'; // or 'AND', based on your use case
      }
      obj.postData.searchLogic = obj.last.logic;
    }
```

You could also "hack" into the search dialog and let the user select the search logic:

```
    onSearchOpen: function(event) {
      event.done(() => {
        var $td = $("td[class=actions]>div").first();
        $td.prepend('Logic: <select><option>AND</option><option>OR</option></select>')
        // todo: set logic on grid when selection changes
      })
    },
```

Fiddle (includes both suggestions): http://jsfiddle.net/gn6cbvf2/

**Comment by vitmalina** [comment]
Argee

**Comment by mpf82** [comment]
Closing as implemented in w2ui 2.0
